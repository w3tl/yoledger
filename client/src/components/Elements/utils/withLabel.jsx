import React from 'react';

const withLabel = Component => ({ id, label = '', ...props }) => {
  const identifier = id || label.toLowerCase();
  if (label) {
    return (
      <label htmlFor={identifier}>
        {`${label}: `}
        <Component id={identifier} {...props} />
      </label>
    );
  }
  return <Component id={identifier} {...props} />;
};

export default withLabel;
