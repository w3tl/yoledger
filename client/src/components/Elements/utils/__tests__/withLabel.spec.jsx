import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import withLabel from '../withLabel';

configure({ adapter: new Adapter() });

function Component() {
  return <input type="text" />;
}

const ComponentWithLabel = withLabel(Component);

describe('TextInput', () => {
  it('should render correctly', () => {
    const output = shallow(
      <ComponentWithLabel id="comp1" label="Label" />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Ñ‹hould substitute a label instead of id if the id is not specified', () => {
    const output = shallow(
      <ComponentWithLabel label="Label" />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
