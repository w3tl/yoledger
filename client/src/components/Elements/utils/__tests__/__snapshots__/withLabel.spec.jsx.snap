// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<label
  htmlFor="comp1"
>
  Label: 
  <Component
    id="comp1"
  />
</label>
`;

exports[`TextInput Ñ‹hould substitute a label instead of id if the id is not specified 1`] = `
<label
  htmlFor="label"
>
  Label: 
  <Component
    id="label"
  />
</label>
`;
